openedx/core/djangoapps/content_libraries/serializers.py

```diff
@@ -190,3 +194,25 @@ class LibraryXBlockStaticFilesSerializer(serializers.Serializer):
     Serializes a LibraryXBlockStaticFile (or a BundleFile)
     """
     files = LibraryXBlockStaticFileSerializer(many=True)
+
+
+class ContentLibraryBlockImportTaskSerializer(serializers.ModelSerializer):
```

Comment:
> Although it is not usual to have hundreds of imports, what do you think about adding pagination to the response?

Thanks for bringing up that discussion.

For this particular model having a hard-coded limit on the number of items returned sorted by most recently created would be cheaper and better aligned with its use-case. Consider that we will have to clean up old entries after some time anyway to avoid cluttering the database with historical information that doesn't have value after some period of time -- something around days would make sense. So, pagination might not be relevant here.

Unless, of course, you have a use case for browsing old import operations. I can foresee, for example, customers wanting to keep certain courses "in sync" with a given library. It might be interesting to keep records of the last import task for that. It could also make sense to make them unique per `(library, course_key)`. I am not a huge fan of doing that because retries might be painful when a task is stuck. If history is a must, I would prefer keeping the last `state != RUNNING` task in the history per pair, and not paginate.

> Could you please add the organization, course ID, and course name to the serializer and response? That way, we would know what course was imported from which organization.

Fair. I'll make the update.

Reply:
> For this particular model having a hard-coded limit on the number of items returned sorted by most recently created would be cheaper and better aligned with its use-case. Consider that we will have to clean up old entries after some time anyway to avoid cluttering the database with historical information that doesn't have value after some period of time -- something around days would make sense. So, pagination might not be relevant here.

I have several concerns about this, though I agree that historical information about import tasks may not so useful in the long run like

* Adding a hardcoded limit for available import tasks (unique per `library` and `course_key`) will result in the need for adjusting the limit, which will lead to PRs for adding a variable to adjust the limit -- not necessarily an issue but worth to note.
* Introducing a hardcoded limit will require some clean-up mechanism, even if it is pre/post save signals, or anything else.

> [...] items returned sorted by most recently created would be cheaper and better aligned with its use-case [...]

Although I'm not a fan of paginated historical import logs either, it is way easier and reliable to keep the import tasks than dealing with their cleanup. Also, the information stored per DB record is really low, we don't have to worry about the 

> [...] I am not a huge fan of doing that because retries might be painful when a task is stuck. If history is a must, I would prefer keeping the last `state != RUNNING` task in the history per pair, and not paginate.

I don't recommend that. That way the users would lose important information, for example, how much time the import failed for a given course in the past 3 days; or even 3 hours.

Reply:
FTR: I added the model and serializer related changes excluding `course_name`. That won't be necessary I believe.

Reply:
Thanks, I've incorporated them into the PR.

I've also included pagination and no clean-up of old records.